app/lib/common/providers/notifiers/sync_notifier.dart

```diff
@@ -48,46 +62,44 @@ class SyncNotifier extends StateNotifier<SyncState> {
 
   void _restartSync() {
     syncState = client.startSync();
-    if (_retryTimer != null) {
-      _retryTimer!.cancel();
-      _retryTimer = null;
-    }
+
+    // reset states
+    _retryTimer?.cancel();
+    _retryTimer = null;
+    _syncPoller?.cancel();
+    _errorPoller?.cancel();
 
     _syncListener = syncState.firstSyncedRx(); // keep it resident in memory
-    _syncPoller = _syncListener.listen((synced) {
+    _syncPoller = _syncListener?.listen((synced) {
       if (synced) {
-        if (mounted) {
-          state = const SyncState(initialSync: false);
-        }
+        state = const SyncState(initialSync: false);
       }
     });
-    ref.onDispose(() => _syncPoller.cancel());
+    ref.onDispose(() => _syncPoller?.cancel());
 
     _errorListener = syncState.syncErrorRx(); // keep it resident in memory
-    _errorPoller = _errorListener.listen((msg) {
+    _errorPoller = _errorListener?.listen((msg) {
       Sentry.captureMessage('Sync failure: $msg', level: SentryLevel.error);
-      if (mounted) {
-        if (msg == 'SoftLogout' || msg == 'Unauthorized') {
```

Comment:
`mounted` isn't necessary anymore as the internal `ref` directly put on the widget is dispose correctly otherwise and so are we.